src/antlr/antlr.test.ts

```diff
@@ -4689,10 +4693,101 @@ describe("antlr", () => {
           }
         }
       ]
+    },
+    {
+      name: "derive fields for unknown tables",
+      sql: "SELECT a_column FROM a_table",
+      data: [
+        {
+          type: "table",
+          id: "result_2",
+          label: "a_table",
+          range: {
+            startLine: 1,
+            startColumn: 21,
+            endLine: 1,
+            endColumn: 28
+          },
+          columns: [
+            {
+              id: "column_1",
+              label: "a_column",
+              isAssumed: true,
+              range: {
+                startLine: 1,
+                startColumn: 7,
+                endLine: 1,
+                endColumn: 15
+              }
+            }
+          ]
+        },
+        {
+          type: "table",
+          id: "result_1",
+          label: "[final result]",
+          range: {
+            startLine: 1,
+            startColumn: 0,
+            endLine: 1,
+            endColumn: 28
+          },
+          columns: [
+            {
+              id: "column_1",
+              label: "a_column",
+              range: {
+                startLine: 1,
+                startColumn: 7,
+                endLine: 1,
+                endColumn: 15
+              }
+            }
+          ]
+        },
+        {
+          type: "edge",
+          edgeType: "select",
+          source: {
+            tableId: "result_2",
+            columnId: "column_1"
+          },
+          target: {
+            tableId: "result_1",
+            columnId: "column_1"
+          }
+        }
+      ]
     }
   ];
 
-  cases.forEach(({ data, name, only, sql, debug, mergedLeaves }) => {
+  function cmpLineage(
+    a: Edge | Table<{ id: string }, { id: string; tableId: string }>,
+    b: Edge | Table<{ id: string }, { id: string; tableId: string }>
+  ): number {
+    switch (a.type) {
+      case "edge":
+        switch (b.type) {
+          case "edge":
+            return 0;
+          case "table":
+            return -1;
+        }
+      case "table":
+        switch (b.type) {
+          case "edge":
+            return 1;
+          case "table":
+            return a.id.localeCompare(b.id);
+        }
+    }
+  }
+
+  cases.forEach(({ data, name, only, sql, debug, mergedLeaves }, idx) => {
+    if (idx === -cases.length - 5) {
```

Comment:
That's an artifact from me trying to run only specific cases. Should not have been in the commit. I'll remove it.

Reply:
BTW, it's very easy to run only specific cases. Just add `only: true` to the case.

Reply:
Well, scrolling in this file isn't fun and I had to run a few of them, so this hack seemed easier ^^.

Reply:
Yeah, I admit that for this file specifically I started to use code folding :)

Reply:
We could rewrite the test logic to load tests from individual json files, each only containing the case and the expectation. Might be much simpler to manage for the future?

Reply:
Yep, we definitely should do something like that. We'll likely get more queries to test, when ppl start using the feature, so having a cleaner setup for this would be good. I'll open a ticket.